tests/unit/moduleapi/propagate.tcl

```diff
@@ -580,6 +599,68 @@ tags "modules" {
                     assert_equal [$replica ttl k1] -1
                 }
 
+                test {module notification on set} {
+                    set repl [attach_to_replication_stream]
+
+                    $master SADD s foo
+
+                    wait_for_condition 500 10 {
+                        [$replica SCARD s] eq "1"
+                    } else {
+                        fail "Failed to wait for set to be replicated"
+                    }
+
+                    $master SPOP s 1
+
+                    wait_for_condition 500 10 {
+                        [$replica SCARD s] eq "0"
+                    } else {
+                        fail "Failed to wait for set to be replicated"
+                    }
+
+                    assert_replication_stream $repl {
+                        {multi}
+                        {select *}
+                        {sadd s foo}
+                        {incr notifications}
+                        {exec}
+                        {multi}
+                        {incr notifications}
+                        {del s}
+                        {incr notifications}
+                        {exec}
+                    }
+                    close_replication_stream $repl
+                }
+
+                test {module key miss notification do not cause read command to be replicated} {
+                    set repl [attach_to_replication_stream]
+
+                    $master flushall
+                    
+                    $master get unexisting_key
+
+                    wait_for_condition 500 10 {
+                        [$replica get missed] eq "1"
+                    } else {
+                        fail "Failed to wait for set to be replicated"
+                    }
+
+                    # Test is checking a wrong!!! behavior that causes a read command to be replicated to replica/aof.
+                    # We keep the test to verify that such a wrong behavior does not cause any crashes.
```

Comment:
ohh, the comment i expected in the module's C code is here..
still maybe drop a comment on the other end too.